pkg/webhooks/webhook_init.go

```diff
@@ -0,0 +1,222 @@
+package webhooks
+
+import (
+	machinev1 "github.com/openshift/api/machine/v1beta1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const (
+	DefaultMachineMutatingHookPath      = "/mutate-machine-openshift-io-v1beta1-machine"
+	DefaultMachineValidatingHookPath    = "/validate-machine-openshift-io-v1beta1-machine"
+	DefaultMachineSetMutatingHookPath   = "/mutate-machine-openshift-io-v1beta1-machineset"
+	DefaultMachineSetValidatingHookPath = "/validate-machine-openshift-io-v1beta1-machineset"
+
+	defaultWebhookConfigurationName = "machine-api"
+	defaultWebhookServiceName       = "machine-api-operator-webhook"
+	defaultWebhookServiceNamespace  = "openshift-machine-api"
+	defaultWebhookServicePort       = 443
+)
+
+var (
+	// webhookFailurePolicy is ignore so we don't want to block machine lifecycle on the webhook operational aspects.
+	// This would be particularly problematic for chicken egg issues when bootstrapping a cluster.
+	webhookFailurePolicy = admissionregistrationv1.Ignore
+	webhookSideEffects   = admissionregistrationv1.SideEffectClassNone
+)
+
+type MAPIWebhookConfig struct {
```

Comment:
I don't like using abbreviations in code and struct names like this, can we call this `MachineAPIWebhookConfig`?